staff/roberto-perez/e-commerce-protos/scripts/cart.js

```diff
@@ -0,0 +1,55 @@
+function Cart() {
+  this.products = [];
+}
+
+Cart.prototype.add = function(product) {
+  if (!(product instanceof Product))
+    throw new TypeError(product + " is not a product");
+
+  return this.products.push(product);
+};
+
+Cart.prototype.totalPrice = function() {
+  return this.products.reduce(function(accum, product) {
+    if (!product.price) product.price = 0;
```

Comment:
In fact an instance of a product should always have price, so in my opinion this validations is unnecessary.

Reply:
Done!